cmd/volume_test.go

```diff
@@ -0,0 +1,181 @@
+			Volume{
+				Volume: hypervisor.Volume{
+					Format: "raw",
+				},
+				SizeMB: 128,
+				Name:   "vol1",
+			},
+			`
+				format: raw
+			`,
+			`
+				file format: raw
+				virtual size: 128M \(134217728 bytes\)
+			`,
+		},
+	}
+	for i, args := range m {
+		c.Logf("CASE #%d: %s", i, args.comment)
+
+		// Prepare.
+		volumesDir := filepath.Join(s.packageDir, "volumes")
+		metaFile := filepath.Join(volumesDir, fmt.Sprintf("vol1.%s.yaml", args.volume.Format))
+		volumeFile := filepath.Join(volumesDir, fmt.Sprintf("vol1.%s", args.volume.Format))
+		ClearDirectory(volumesDir)
+		PrepareFiles(s.packageDir, map[string]string{"/meta/package.yaml": PackageYamlText})
+
+		// This is what we're testing here.
+		err := CreateVolume(s.packageDir, args.volume)
+
+		// Expectations.
+		c.Assert(err, IsNil)
+		c.Check(metaFile, FileMatches, FixIndent(args.expectedMeta))
+		c.Check([]string{"qemu-img", "info", volumeFile}, CmdOutputMatches, FixIndent(args.expectedInfo))
+	}
+}
+
+func (s *volumesSuite) TestCreateVolumeInvalidNoPackage(c *C) {
+	// This is what we're testing here.
+	err := CreateVolume(s.packageDir, Volume{})
+
+	// Expectations.
+	c.Check(err, ErrorMatches, "Must be in package root directory")
+
+}
+
+func (s *volumesSuite) TestDeleteVolume(c *C) {
+	m := []struct {
+		comment         string
+		state           map[string]string
+		name            string
+		expectedVolumes map[string]interface{}
+	}{
+		{
+			"simple",
+			map[string]string{
+				"/volumes/vol1.qcow2":      DefaultText,
+				"/volumes/vol1.qcow2.yaml": DefaultText,
+			},
+			"vol1",
+			map[string]interface{}{},
+		},
+		{
+			"deletes all formats",
+			map[string]string{
+				"/volumes/vol1.qcow2":      DefaultText,
+				"/volumes/vol1.qcow2.yaml": DefaultText,
+				"/volumes/vol1.raw":        DefaultText,
+				"/volumes/vol1.raw.yaml":   DefaultText,
+			},
+			"vol1",
+			map[string]interface{}{},
+		},
+		{
+			"delete unexisting",
+			map[string]string{
+				"/volumes/vol1.qcow2":      DefaultText,
+				"/volumes/vol1.qcow2.yaml": DefaultText,
+			},
+			"unexisting-volume",
+			map[string]interface{}{
+				"vol1.qcow2":      DefaultText,
+				"vol1.qcow2.yaml": DefaultText,
+			},
+		},
+		{
+			"don't delete same prefix",
+			map[string]string{
+				"/volumes/vol1.qcow2":           DefaultText,
+				"/volumes/vol1.qcow2.yaml":      DefaultText,
+				"/volumes/vol1-suffix.raw":      DefaultText,
+				"/volumes/vol1-suffix.raw.yaml": DefaultText,
+			},
+			"vol1",
+			map[string]interface{}{
+				"vol1-suffix.raw":      DefaultText,
+				"vol1-suffix.raw.yaml": DefaultText,
+			},
+		},
+	}
+	for i, args := range m {
+		c.Logf("CASE #%d: %s", i, args.comment)
+
+		// Prepare.
+		volumesDir := filepath.Join(s.packageDir, "volumes")
+		args.state["/meta/package.yaml"] = PackageYamlText
+		PrepareFiles(s.packageDir, args.state)
+
+		// This is what we're testing here.
+		err := DeleteVolume(s.packageDir, args.name)
+
+		// Expectations.
+		c.Assert(err, IsNil)
```

Comment:
If you decide to implement errors in case of non-existing volumes, don't forget to modify assertions here.